language/move-prover/stackless-bytecode-generator/src/lifetime_analysis.rs

```diff
@@ -8,18 +8,63 @@ use vm::file_format::CodeOffset;
 
 use crate::{
     dataflow_analysis::{DataflowAnalysis, StateMap, TransferFunctions},
+    function_target::FunctionTargetData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
     stackless_bytecode::{
         AssignKind,
         Bytecode::{self, *},
         TempIndex,
     },
     stackless_control_flow_graph::StacklessControlFlowGraph,
 };
-use spec_lang::ty::Type;
+use spec_lang::{env::FunctionEnv, ty::Type};
+
+/// Function target annotation computed by the lifetime analysis processor.
+pub struct LifetimeAnnotation(BTreeMap<CodeOffset, BTreeSet<TempIndex>>);
+
+impl LifetimeAnnotation {
+    pub fn get_dead_refs(&self, code_offset: CodeOffset) -> Option<&BTreeSet<TempIndex>> {
```

Comment:
Do reference issues force us into using  `Option<&BTreeSet<TempIndex>>` instead of `&BTreeSet<TempIndex>` here?

Reply:
This is simply the convention of Rust's `collection.get` API. The reference is inside the option. also makes sense for unpacking the thing, which you do not want to move.